### DISCUSSION TITLE
Unable to add tracker

### DISCUSSION THREAD

**Discussion by Sice-0** [transmission/transmission#8513] [Support]
### What is the issue?

   When adding many (it can be more than 300) tracker servers to the tracker list in the torrent  , **_"No new URLs found."_** will be returned. 
   This issue also happends with the default public trackerlist setting introduced in version 4.0, where even if you enter a large number of tracker servers in the settings, it will not be added to the trackerlist of new torrents for you.
   I believe this happens with any torrent. 

I know adding a lot of trackers to the torrent is unnecessary, but it returns the wrong message, so I think it might be a bug. 

### Which application of Transmission?

Qt app on Windows

### Which version of Transmission?

4.0.4

**Comment by killemov** [comment]
I have not tested the max. number of torrents. But I'd guess that it would be a power of 2. (Or that -1.) As for the default public tracker list, is that list appended to the torrent trackers upon adding the torrent?

**Comment by Sice-0** [comment]
> I have not tested the max. number of torrents. But I'd guess that it would be a power of 2. (Or that -1.) As for the default public tracker list, is that list appended to the torrent trackers upon adding the torrent?

It works fine when there are only a few trackers in the default public tracker list. But as I said, once the number is high (like 300) it doesn't work properly, and it doesn't add any tracker to the newly added torrents.

**Comment by Sice-0** [comment]
Sorry I accidentally closed it :)

**Comment by ckerr** [comment]
The string `The correct URL was not found` doesn't seem to be anywhere in the codebase. Are you certain that's the error message?

**Comment by Sice-0** [comment]
> The string `The correct URL was not found` doesn't seem to be anywhere in the codebase. Are you certain that's the error message?

Thank you for your reply, Mr. ckerr.
Sorry I didn't know what the exact English message is because I can't set the language to English...
I just confirmed it and successfully reproduced the problem on another windows10_22h2.
The exact message is 

> **No new URLs found.**

![no-new-urls-found](https://github.com/transmission/transmission/assets/134118047/0bdbd7a8-5cdd-4c17-a77b-636a7aaac4eb)
And if necessary, I can post the trackerlist used for testing.

**Comment by ckerr** [comment]
Yes, please do post the trackerlist :+1:

**Comment by Sice-0** [comment]
> Yes, please do post the trackerlist 👍

https://nyaa.tracker.wf:7777/announce
http://rotracker.ohys.net:80/announce
http://ehtracker.org/1/announce
http://171.104.111.14:6969/announce
http://83.31.209.3:6969/announce
http://104.244.77.14:1337/announce
http://thetracker.org/announce
http://83.31.199.157:6969/announce
http://tracker.nyacat.pw:7000/announce.php
http://www.siambt.com/announce.php
http://60-fps.org:80/bt/announce.php
http://xtremewrestlingtorrents.net:80/announce.php
http://bt.beatrice-raws.org:80/announce
http://tracker.trancetraffic.com:80/announce.php
http://mixfiend.com:80/announce.php
http://163.172.29.130:80/announce
http://alltorrents.net:80/bt/announce.php
http://announce.partis.si:80/announce
http://mvgforumtracker.mvgroup.org:80/tracker.php/announce
http://156.234.201.18:80/announce
https://t.zerg.pw/announce
http://frp.v2fy.com:8000/announce
https://tracker.renfei.net/announce
http://rotracker.ohys.net/announce
udp://198.100.149.66:6969/announce
udp://109.201.134.183:80/announce
udp://178.170.48.154:1337/announce
http://185.197.195.20:1919/announce
http://83.31.191.97:6969/announce
http://45.154.253.4:80/announce
http://171.104.226.15:6969/announce
http://83.31.219.73:6969/announce
udp://ln.mtahost.co:6969/announce
http://tracker.aibt.xyz:900/announce
http://171.104.226.31:6969/announce
http://97.117.85.73:9000/announce
http://www.bit-hdtv.com:2710/announce
https://tk.mabo.ltd:443/announce
udp://172.105.235.127:6969/announce
udp://103.122.21.50:6969/announce
udp://173.249.201.201:6969/announce
udp://104.131.98.232:6969/announce
udp://209.126.11.233:6969/announce
udp://192.95.46.115:6969/announce
udp://167.99.185.219:6969/announce
udp://51.15.3.74:6969/announce
udp://51.15.26.25:6969/announce
udp://51.158.144.42:6969/announce
udp://144.91.88.22:6969/announce
udp://51.159.54.68:6666/announce
udp://163.172.29.130:80/announce
udp://88.99.2.212:6969/announce
udp://148.251.53.72:6969/announce
udp://193.37.214.12:6969/announce
udp://82.65.115.10:6969/announce
udp://37.187.111.136:6969/announce
udp://176.31.250.174:6969/announce
udp://5.196.89.204:6969/announce
udp://5.196.67.51:6969/announce
udp://37.59.48.81:6969/announce
udp://37.187.95.112:6969/announce
udp://161.97.67.210:6969/announce
udp://185.102.219.163:6969/announce
udp://95.216.74.39:6969/announce
udp://135.125.106.92:6969/announce
udp://217.30.10.77:6969/announce
udp://93.104.214.40:6969/announce
udp://51.68.174.87:6969/announce
udp://45.9.60.30:6969/announce
udp://158.101.161.60:3131/announce
udp://94.103.87.87:6969/announce
udp://94.243.222.100:6969/announce
udp://91.211.5.21:6969/announce
udp://88.80.28.7:6969/announce
udp://194.38.21.77:6969/announce
udp://95.31.11.224:6969/announce
udp://46.138.242.240:6969/announce
http://171.104.111.20:6969/announce
http://97.117.114.88:9000/announce
udp://51.81.222.188:6969/announce
http://tr.nyacat.pw/announce
http://tracker.btcake.com/announce
https://tr.nyacat.pw/announce
http://tracker.btcake.com:80/announce
http://tracker.peckservers.com:9000/announce
https://tracker.bangumi.zip/announce
https://tracker.ipfsscan.io/announce
http://shogiroom.com:6969/announce
udp://107.175.221.194:6969/announce
http://www.bitseduce.com/announce.php
http://houniao.ddns.net:8888/announce
udp://open.publictracker.xyz:6969/announce
https://t.zerg.pw:443/announce
http://thitgaluoc.dynu.net:6969/announce
http://smurfsoft.com:6969/announce
http://region.nl1.privex.cc:6969/announce
http://canardscitrons.nohost.me:6969/announce
http://t.publictracker.xyz:6969/announce
http://li2021-95.members.linode.com:6969/announce
http://mail.lakameraobscura.com:6969/announce
http://unknownsite.de:6969/announce
http://wg.mortis.me:6969/announce
http://www.chouchou.club:8080/announce
udp://t.zerg.pw:6969/announce
http://ipv4.tracker.harry.lu/announce
https://tr.torland.ga/announce
https://tr.torland.ga:443/announce
https://tk.mabo.ltd/announce
http://tk2.greedland.net/announce
http://tracker.bt-chat.com/announce
http://torrents.linuxmint.com:80/announce.php
http://www.megatorrents.kg:80/announce.php
http://torrent.unix-ag.uni-kl.de:80/announce
udp://185.230.4.150:1337/announce
http://185.230.4.150:1337/announce
http://ehtracker.org/2496841/announce
http://home.kray.pw:6969/announce
http://p4p.arenabg.com:1337/announce
https://tracker.cloudit.top/announce
http://45.154.253.10:80/announce
http://shubt.net:2710/announce
http://tracker.ex.ua/announce
udp://tracker.ex.ua:80/announce
http://tracker.ex.ua:80/announce

Sorry I closed it again. . This is my first time using GitHub and I'm still learning.

**Comment by Sice-0** [comment]
Please help me check whether the problem lies in an unsupported tracker type or the wrong one
url format, thank you very much.   m(._.)m

**Comment by ckerr** [comment]
Looks like one problem is you're including http://rotracker.ohys.net:80/announce in there twice.

The error message could be more helpful though!

**Comment by Pentaphon** [comment]
@Sice-0 that is a really huge list. I wouldn't recommend using that many trackers. In the future you should just use the "Best" list of 20 trackers from https://github.com/ngosang/trackerslist which is what many people are doing these days for your usecase. This list + DHT should be enough for everybody on public torrents.

**Comment by Sice-0** [comment]
> Looks like one problem is you're including http://rotracker.ohys.net:80/announce in there twice.
> 
> The error message could be more helpful though!

Thanks, I found this out with your help. But the message _**"No new URLs found."**_ will still be returned, after I delete 

> http://rotracker.ohys.net:80/announce

**Comment by Sice-0** [comment]
> @Sice-0 that is a really huge list. I wouldn't recommend using that many trackers. In the future you should just use the "Best" list of 20 trackers from https://github.com/ngosang/trackerslist which is what many people are doing these days for your usecase. This list + DHT should be enough for everybody on public torrents.

Thanks for your reply, Pentaphon. This is indeed a very huge list (≧∇≦). 
And thanks for the advice, I will try to only use the "Best" list. But I think maybe we can do something, like add support for a huge number of trackerlists, or return **"_Exceed the maximum_"** when adding a huge number of trackers”.
